pouch/convertor/cri.go

```diff
@@ -0,0 +1,225 @@
+	if err != nil {
+		return nil, err
+	}
+	sandbox.Config = sandboxConfig
+
+	return sandbox, nil
+}
+
+// type PodSandboxConfig struct {
+//     Metadata *PodSandboxMetadata
+//     Hostname string
+//     LogDirectory string
+//     DnsConfig *DNSConfig
+//     PortMappings []*PortMapping
+//     Labels map[string]string
+//     Annotations map[string]string
+//     Linux *LinuxPodSandboxConfig
+// }
+//
+func toPodSandboxConfig(c *localtypes.Container) (*runtime.PodSandboxConfig, error) {
+	sandboxConfig := &runtime.PodSandboxConfig{
+		Labels: map[string]string{},
+	}
+
+	meta, err := getPodSandboxMetadataBySandboxName(c.Name)
+	if err != nil {
+		return nil, err
+	}
+	sandboxConfig.Metadata = meta
+
+	if c.Config != nil {
+		sandboxConfig.Hostname = string(c.Config.Hostname)
+		sandboxConfig.Annotations = c.Config.SpecAnnotation
+
+		for k, v := range c.Config.Labels {
+			// filter annotation labels
+			if !strings.HasPrefix(k, "annotation.") && !utils.StringInSlice([]string{PouchContainerTypeLabelKey, DockerContainerTypeLabelKey}, k) {
+				sandboxConfig.Labels[k] = v
+			}
+		}
+	}
+
+	// LogDirectory
+	sandboxConfig.LogDirectory = fmt.Sprintf("/var/log/pods/%s", meta.Uid)
+
+	// Linux
+	linuxConfig, err := toLinuxPodSandboxConfig(c)
+	if err != nil {
+		return nil, err
+	}
+	sandboxConfig.Linux = linuxConfig
+
+	// DnsConfig
+	dnsConf, err := toDNSConfig(c.ResolvConfPath)
+	if err != nil {
+		return nil, err
+	}
+	sandboxConfig.DnsConfig = dnsConf
+
+	return sandboxConfig, nil
+}
+
+func getPodSandboxMetadataBySandboxName(name string) (*runtime.PodSandboxMetadata, error) {
+	if !SandboxNameRegex.Match([]byte(name)) {
+		return nil, fmt.Errorf("name %s not match k8s sandbox name regexp %s", name, SandboxNameRegex.String())
+	}
+
+	parts := strings.Split(name, "_")
+	attempt, err := strconv.ParseUint(parts[5], 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse the attempt times in sandbox name: %q: %v", name, err)
+	}
+
+	return &runtime.PodSandboxMetadata{
+		Name:      parts[2],
+		Namespace: parts[3],
+		Uid:       parts[4],
+		Attempt:   uint32(attempt),
+	}, nil
+}
+
+func toLinuxPodSandboxConfig(c *localtypes.Container) (*runtime.LinuxPodSandboxConfig, error) {
+	linuxConfig := &runtime.LinuxPodSandboxConfig{}
+	if c.HostConfig != nil {
+		linuxConfig.CgroupParent = c.HostConfig.CgroupParent
+		linuxConfig.Sysctls = c.HostConfig.Sysctls
+	}
+
+	linuxSecurityContext, err := toLinuxSandboxSecurityContext(c)
+	if err != nil {
+		return nil, err
+	}
+	linuxConfig.SecurityContext = linuxSecurityContext
+
+	return linuxConfig, nil
+}
+
+func toLinuxSandboxSecurityContext(c *localtypes.Container) (*runtime.LinuxSandboxSecurityContext, error) {
+	linuxSecurityContext := &runtime.LinuxSandboxSecurityContext{
+		NamespaceOptions: &runtime.NamespaceOption{
+			Pid: runtime.NamespaceMode_CONTAINER,
+		},
+	}
+	if c.NetworkSettings != nil {
+		if _, ok := c.NetworkSettings.Networks["host"]; ok {
```

Comment:
Is it appropriate to set all to host mode according to the NetworkSettings？

Reply:
actually, we indeed set all to host mode if the pod is a daemonset.